documentation/src/docs/basic-reader-and-writer.md

```diff
@@ -331,3 +331,21 @@ while (events.hasNext())
 }
 
 ```
+getNextStreamCut API:  For a streaming application like Spark, which uses micro-batch reader connectors to read Pravega Streams in parallel batches, needs a streamCut till what the batch can read.
+This api provides a streamCut that is in a location that has not yet been read but is at a bounded distance from the reader's current location. This api takes a starting streamCut and an approximate distance in bytes and return a new stream cut. No segments from the starting streamCut is skipped over. So the cut is either present inside the segments of the starting streamCut or at the boundry of the segments and it may advance to the successor segments if it is at the tail.
+```Java
+StreamCut startingStreamCut = streamManager.fetchStreamInfo(streamScope, streamName).join().getHeadStreamCut();
+long approxDistanceToNextOffset = 350L;
+StreamCut nextStreamCut = client.getNextStreamCut(startingStreamCut, approxDistanceToNextOffset);
+```
+- Depending on the approximate distance per number of segments in the current streamcut, next offset for each segment is requested. (For example - If the startingStreamCut is containing two segments 1 and 2 and the approxDistanceToNextOffset is sent as 100, then the next offset in segment 1 and segment 2 is searched with a value of 50 bytes each).
```

Comment:
Again very small example numbers is both unrealistic as an example and implies a precision to the API that is simply not there. The location is approximate.